bin/setupReplication.js

```diff
@@ -0,0 +1,449 @@
+const async = require('async');
+const commander = require('commander');
+const { S3, IAM, SharedIniFileCredentials } = require('aws-sdk');
+
+const { Logger } = require('werelogs');
+
+const config = require('../conf/Config');
+
+const trustPolicy = {
+    Version: '2012-10-17',
+    Statement: [
+        {
+            Effect: 'Allow',
+            Principal: {
+                Service: 'backbeat',
+            },
+            Action: 'sts:AssumeRole',
+        },
+    ],
+};
+
+function _buildResourcePolicy(source, target) {
+    return {
+        Version: '2012-10-17',
+        Statement: [
+            {
+                Effect: 'Allow',
+                Action: [
+                    's3:GetObjectVersion',
+                    's3:GetObjectVersionAcl',
+                ],
+                Resource: [
+                    `arn:aws:s3:::${source}/*`,
+                ],
+            },
+            {
+                Effect: 'Allow',
+                Action: [
+                    's3:ListBucket',
+                    's3:GetReplicationConfiguration',
+                ],
+                Resource: [
+                    `arn:aws:s3:::${source}`,
+                ],
+            },
+            {
+                Effect: 'Allow',
+                Action: [
+                    's3:ReplicateObject',
+                    's3:ReplicateDelete',
+                ],
+                Resource: `arn:aws:s3:::${target}/*`,
+            },
+        ],
+    };
+}
+
+function _setupS3Client(host, port, profile) {
+    const credentials = new SharedIniFileCredentials({ profile });
+    return new S3({
+        endpoint: `http://${host}:${port}`,
+        sslEnabled: false,
+        credentials,
+        s3ForcePathStyle: true,
+        region: 'file',
+    });
+}
+
+
+function _setupIAMClient(host, port, profile) {
+    const credentials = new SharedIniFileCredentials({ profile });
+    return new IAM({
+        endpoint: `http://${host}:${port}`,
+        sslEnabled: false,
+        credentials,
+        maxRetries: 0,
+        region: 'file',
+        signatureCache: false,
+        s3ForcePathStyle: true,
+        httpOptions: {
+            timeout: 1000,
+        },
+    });
+}
+
+class _SetupReplication {
+    /**
+     * This class sets up two buckets for replication.
+     * @constructor
+     * @param {String} sourceBucket - Source Bucket Name
+     * @param {String} targetBucket - Destination Bucket Name
+     * @param {Object} log - Werelogs Request Logger object
+     * @param {Object} config - bucket configurations
+     */
+    constructor(sourceBucket, targetBucket, log, config) {
+        const { source, destination } = config.extensions.replication;
+        this._log = log;
+        this._sourceBucket = sourceBucket;
+        this._targetBucket = targetBucket;
+        this._s3 = {
```

Comment:
Since this is for multiple s3 clients perhaps a name like `this._s3Clients` is suitable. That would also apply to `this._iam`.